builder/azure/arm/clientconfig.go

```diff
@@ -0,0 +1,197 @@
+package arm
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	jwt "github.com/dgrijalva/jwt-go"
+	packerAzureCommon "github.com/hashicorp/packer/builder/azure/common"
+	"github.com/hashicorp/packer/packer"
+)
+
+// ClientConfig allows for various ways to authenticate Azure clients
+type ClientConfig struct {
+	// Describes where API's are
```

Comment:
Where's the description?

Reply:
in the CloudEnvironment.... it lists a bunch of urls, where the Azure API's are...